src/v3/src/components/Widget/index.tsx

```diff
@@ -210,7 +212,10 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const handleError = (error: unknown) => {
+  const handleError = async (error: unknown) => {
+    if (widgetHooks) {
+      await widgetHooks.callHooks('before', undefined, error as Error);
+    }
```

Comment:
Anywhere we are doing a `if(widgetHooks)` guard we can take advantage of optional chaining syntax:
```suggestion
    await widgetHooks?.callHooks('before', undefined, error as Error);
```

Reply:
We could also make `widgetHooks` a required prop

Reply:
I made `widgetHooks` a required prop and pass it to `<Widget>` in [`setup()`](https://github.com/okta/okta-signin-widget/blob/master/src/v3/test/integration/util/setup.tsx#L43) for tests
Otherwise there will be error `Cannot read properties of undefined (reading 'callHooks')`